### PR TITLE
SLI-1055 Expose Clean Code Taxonomy fields on Intellij

### PR DESCRIPTION
**Pull request by eray-felek-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/finding/LiveFinding.java [hunk 1]
```diff
@@ -69,6 +76,17 @@ protected LiveFinding(Issue issue, PsiFile psiFile, @Nullable RangeMarker range,
...
     this.ruleDescriptionContextKey = issue.getRuleDescriptionContextKey().orElse(null);
+    this.cleanCodeAttribute = issue.getCleanCodeAttribute();
+
+    if(!issue.getImpacts().isEmpty()){
+      this.impacts = issue.getImpacts();
+    } else {
+      Map<SoftwareQuality, ImpactSeverity> defaultImpacts = new HashMap<>();
+      defaultImpacts.put(SoftwareQuality.MAINTAINABILITY, ImpactSeverity.HIGH);
+      defaultImpacts.put(SoftwareQuality.RELIABILITY, ImpactSeverity.MEDIUM);
+      defaultImpacts.put(SoftwareQuality.SECURITY, ImpactSeverity.LOW);
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/finding/LiveFinding.java:87]: Is this just some mocking to test?

File: src/main/java/org/sonarlint/intellij/finding/LiveFinding.java [hunk 2]
```diff
@@ -69,6 +76,17 @@ protected LiveFinding(Issue issue, PsiFile psiFile, @Nullable RangeMarker range,
     this.context = context;
     this.quickFixes = quickFixes;
     this.ruleDescriptionContextKey = issue.getRuleDescriptionContextKey().orElse(null);
+    this.cleanCodeAttribute = issue.getCleanCodeAttribute();
+
+    if(!issue.getImpacts().isEmpty()){
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/finding/LiveFinding.java:81]: Please make sure you use the Sonar code style in your IDE

File: src/main/java/org/sonarlint/intellij/finding/LiveFinding.java [hunk 3]
```diff
@@ -69,6 +76,17 @@ protected LiveFinding(Issue issue, PsiFile psiFile, @Nullable RangeMarker range,
...
+      this.impacts = issue.getImpacts();
+    } else {
+      Map<SoftwareQuality, ImpactSeverity> defaultImpacts = new HashMap<>();
+      defaultImpacts.put(SoftwareQuality.MAINTAINABILITY, ImpactSeverity.HIGH);
+      defaultImpacts.put(SoftwareQuality.RELIABILITY, ImpactSeverity.MEDIUM);
+      defaultImpacts.put(SoftwareQuality.SECURITY, ImpactSeverity.LOW);
+      this.impacts = defaultImpacts;
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/finding/LiveFinding.java:88]: I think you can just do `this.impacts = Map.of(SoftwareQuality.MAINTAINABILITY, ImpactSeverity.HIGH, SoftwareQuality.RELIABILITY, ImpactSeverity.MEDIUM, SoftwareQuality.SECURITY, ImpactSeverity.LOW);`

File: src/main/java/org/sonarlint/intellij/finding/LiveFinding.java (hunk 1 above)
Comment (eray-felek-sonarsource) [reply src/main/java/org/sonarlint/intellij/finding/LiveFinding.java:87]: Yes

File: src/main/java/org/sonarlint/intellij/finding/LiveFinding.java [hunk 4]
```diff
@@ -69,6 +75,13 @@ protected LiveFinding(Issue issue, PsiFile psiFile, @Nullable RangeMarker range,
     this.context = context;
     this.quickFixes = quickFixes;
     this.ruleDescriptionContextKey = issue.getRuleDescriptionContextKey().orElse(null);
+    this.cleanCodeAttribute = issue.getCleanCodeAttribute();
+
+    if (!issue.getImpacts().isEmpty()) {
+      this.impacts = issue.getImpacts();
+    } else {
+      this.impacts = Map.of(SoftwareQuality.MAINTAINABILITY, ImpactSeverity.HIGH, SoftwareQuality.RELIABILITY, ImpactSeverity.MEDIUM, SoftwareQuality.SECURITY, ImpactSeverity.LOW);
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/finding/LiveFinding.java:83]: Last point, could you add a TODO comment just so we don't forget? 😄

**Review by nquinquenel — APPROVED** [review]
👍🏻

File: src/test/java/org/sonarlint/intellij/finding/issue/IssueFixtures.kt [hunk 5]
```diff
@@ -23,22 +23,27 @@ import com.intellij.openapi.editor.Document
...
 import org.sonarsource.sonarlint.core.commons.TextRange
 import org.sonarsource.sonarlint.core.commons.VulnerabilityProbability
+import java.util.Map
 import java.util.Optional
 
 fun aLiveIssue(
     file: PsiFile,
     rangeMarker: RangeMarker? = file.virtualFile.getDocument()!!.createRangeMarker(0, 1),
-    coreIssue: Issue = aCoreIssue(file, toTextRange(rangeMarker))
+    coreIssue: Issue = aCoreIssue(file, toTextRange(rangeMarker)),
```
Comment (nquinquenel) [inline src/test/java/org/sonarlint/intellij/finding/issue/IssueFixtures.kt:46]:
```suggestion
    coreIssue: Issue = aCoreIssue(file, toTextRange(rangeMarker))
```

**Review by nquinquenel — APPROVED** [review]

File: src/test/java/org/sonarlint/intellij/finding/issue/IssueFixtures.kt [hunk 6]
```diff
@@ -54,6 +59,19 @@ fun aCoreIssue(file: PsiFile, textRange: TextRange? = TextRange(0, 0, 0, 1)) = o
     override fun getType() = RuleType.BUG
     override fun getRuleKey() = "ruleKey"
     override fun flows() = mutableListOf<Flow>()
+    override fun getCleanCodeAttribute() = CleanCodeAttribute.defaultCleanCodeAttribute()
+
+    override fun getImpacts(): MutableMap<SoftwareQuality, ImpactSeverity> {
+        return Map.of(
```
Comment (nquinquenel) [inline src/test/java/org/sonarlint/intellij/finding/issue/IssueFixtures.kt:65]: Also, use kotlin map and not java map

`        return mutableMapOf(
            SoftwareQuality.MAINTAINABILITY to ImpactSeverity.HIGH,
            SoftwareQuality.RELIABILITY to ImpactSeverity.MEDIUM,
            SoftwareQuality.SECURITY to ImpactSeverity.LOW
        )`

**Pull request merged**